src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        BulldozerProducer bulldozerProducer = new BulldozerProducer();
+        ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+        TruckProducer truckProducer = new TruckProducer();
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(bulldozerProducer.get());
+        }
+        if (type == Excavator.class) {
+            return new ArrayList<>(excavatorProducer.get());
+        }
+        if (type == Truck.class) {
+            return new ArrayList<>(truckProducer.get());
+        }
+        return new ArrayList<>();
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        int count = machines.size();
+        machines.clear();
+        for (int i = 0; i < count; i++) {
+            machines.add(i, value);
```

Comment:
```suggestion
        for (int i = 0; i < machines.size(); i++) {
            machines.add(i, value);
```

Reply:
If I do this Suggested I will have an infinite loop, because machines.size always increase. I do like this:
<img width="558" alt="image" src="https://user-images.githubusercontent.com/98731880/170727403-fc01de74-3899-45ef-a6cc-db62df5b9880.png">

Reply:
Це мені вже треба відпочити)) оце я дав, аж стидно)